api/utils/keys/privatekey.go

```diff
@@ -61,32 +61,24 @@ type PrivateKey struct {
 	keyPEM []byte
 }
 
-// NewPrivateKey returns a new PrivateKey for the given crypto.Signer with a
-// pre-marshaled private key PEM, which may be a special PIV key PEM.
-func NewPrivateKey(signer crypto.Signer, keyPEM []byte) (*PrivateKey, error) {
-	sshPub, err := ssh.NewPublicKey(signer.Public())
+// NewPrivateKey returns a new PrivateKey for a crypto.Signer.
+// [signer] must be an *rsa.PrivateKey, *ecdsa.PrivateKey, ed25519.PrivateKey, or *hardwarekey.PrivateKey.
+// TODO(Joerger): Remove the variadic argument once /e is updated to not provide it.
+func NewPrivateKey(signer crypto.Signer, _ ...[]byte) (*PrivateKey, error) {
```

Comment:
note: it would be nice if we actually didn't have to marshal the key to PEM when a lot of the time it is not even necessary and just wastes CPU cycles when we never write out the key. The fact PrivateKeyPEM() currently doesn't return an error is what locks us in I guess we'd have to update every caller. Not expecting you to do anything about this here just calling it out